### ISSUE TITLE
IDR rendering module

### ISSUE THREAD

**Issue by ankarako** [Kai-46/PhySG#15]
Hi and congratulations for your interesting work!

I would like to ask you about the participation of the IDR rendering module in the optimization process. From reading the paper I assumed that material properties (including the diffuse albedo) are estimated by the  MaterialEnvmap module you introduce in your paper.
However, in your [code](https://github.com/Kai-46/PhySG/blob/030d590bc007377007c65046c8f70ad149cfb371/code/model/implicit_differentiable_renderer.py#L323) , you utilize both your module and IDR's rendering module (which both estimate rgb values basically). In addition, color values calculated by both IDR's rendering module and the MaterialEnvmap module participate in the [loss calculation](https://github.com/Kai-46/PhySG/blob/030d590bc007377007c65046c8f70ad149cfb371/code/model/loss.py#L33).

Do you train both modules (IDR Rendering and MaterialEnvmap) for a specific reason, or just for comparison?

PS: In addition, I can't seem to find if EnvmapMaterial module's weights are initialized. Could you help?
Thanks for your time.

**Comment by Kai-46** [comment]
The weight for that idr rendering loss is set to zero in the config. Hence it doesn't matter.

**Comment by ankarako** [comment]
Thanks I didn't notice that :). What about weight initialization for the EnvmapMaterial module

**Comment by Kai-46** [comment]
Hi --- the weight initialization for EnvmapMaterial are all inside this script: https://github.com/Kai-46/PhySG/blob/030d590bc007377007c65046c8f70ad149cfb371/code/model/sg_envmap_material.py#L46

**Comment by ankarako** [comment]
Thanks, I saw that, but I was a little bit curious about the weight initialization of the network's linear layers. I guess you use pytorch's default initialization (which must be xavier if I remember correctly).

Thanks again.
